client/network/src/protocol/sync.rs

```diff
@@ -1047,6 +1057,24 @@ impl<B: BlockT> ChainSync<B> {
 		Ok(OnBlockJustification::Nothing)
 	}
 
+	pub fn register_preimport_blocks(&self, blocks: Vec<message::BlockData<B>>) {
+		self.verified_blocks.extend(blocks.clone().into_iter().filter_map(|b| {
```

Comment:
```suggestion
		self.verified_blocks.extend(blocks.into_iter().filter_map(|b| {
```

clone makes no sense.